### PR TITLE
*: Support `PLACEMENT_ADMIN` dynamic priv

### LINKED ISSUE #27603: Ensure {CREATE|ALTER|DROP} PLACEMENT POLICY requires the PLACEMENT_ADMIN privilege
This is a sub-task of Placement Rules in SQL: https://github.com/pingcap/tidb/issues/18030

Support the ability to `CREATE PLACEMENT POLICY` with the same options available in `CREATE TABLE|ALTER TABLE` context.

The following subtasks are known:

* [ ] Ensure {CREATE|ALTER|DROP} PLACEMENT POLICY requires the PLACEMENT_ADMIN privilege
    * [ ] Require CREATE PLACEMENT POLICY with `PLACEMENT_ADMIN`
    * [ ] Require DROP PLACEMENT POLICY with `PLACEMENT_ADMIN`
    * [ ] Require ALTER PLACEMENT POLICY with `PLACEMENT_ADMIN`

### PR DESCRIPTION
**Pull request by sylzd**
### What problem does this PR solve?

Issue Number: fixes #27603 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Check List 

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
Please add a release note, or a 'None' if it is not needed.
```

### CODE REVIEW

**Comment by morgo** [comment]
@sylzd This covers the first part (adding the permission). Be sure to  [add the privilege check with `visitInfo`](https://pingcap.github.io/tidb-dev-guide/understand-tidb/privilege.html#adding-privilege-checks-to-a-statement).

**Comment by sylzd** [comment]
> @sylzd This covers the first part (adding the permission). Be sure to [add the privilege check with `visitInfo`](https://pingcap.github.io/tidb-dev-guide/understand-tidb/privilege.html#adding-privilege-checks-to-a-statement).

thanks, I  am working on it.

**Review by morgo — APPROVED** [review]

File: privilege/privileges/privileges_test.go [hunk 1]
```diff
@@ -2299,3 +2299,58 @@ func TestShowGrantsForCurrentUserUsingRole(t *testing.T) {
...
+	defer clean()
+	se := newSession(t, store, dbName)
+	mustExec(t, se, "drop placement policy if exists x")
+	createStmt := "create placement policy x " +
+		"PRIMARY_REGION=\"cn-east-1\" " +
+		"REGIONS=\"cn-east-1,cn-east-2\" " +
+		"LEARNERS=1 " +
+		"LEARNER_CONSTRAINTS=\"[+region=cn-west-1]\" " +
+		"VOTERS=3 " +
+		"VOTER_CONSTRAINTS=\"[+disk=ssd]\""
```
Comment (xhebox) [inline privilege/privileges/privileges_test.go:2337]: Maybe remove these lines. It is better to be simple: syntax checks are added such that your statement failed.

**Review by xhebox — APPROVED** [review]

**Comment by AilinKid** [comment]
CLA is hanged, you can push a empty commit to trigger it again

**Comment by sylzd** [comment]
@xhebox could you plz help to merge... again?

**Pull request merged**